File: Graph_Traversal/BOJ_2606(바이러스_bfs).py
#BFS 풀이
from collections import deque
n=int(input())
m=int(input())
graph=[[]*i for i in range(n+1)]
visited=[False]*(n+1)
answer=0
for _ in range(m): #그래프 생성
    a,b=map(int,input().split())
    graph[a].append(b)
    graph[b].append(a)
    graph[a].sort()
    graph[b].sort()
def bfs(graph,visited,answer):
    que=deque()
    que.append(1)
    visited[1]=True
    while que:
        cur=que.popleft()
        if visited[cur]==False:
            visited[cur]=True
            answer+=1
        for i in graph[cur]:
            if visited[i]==False:
                que.append(i)
                visited[i]=True
                answer+=1
    return answer
print(bfs(graph,visited,answer))

File: Graph_Traversal/BOJ_2606(바이러스_bfs).cpp
#include <iostream>
#include <vector>
#include <queue>
#include <algorithm>
#define MAX 101
using namespace std;

vector<int> graph[MAX];
bool visited[MAX];
queue<int> que;
int n, m, k, a, b;
int cnt = 0;

int bfs(int start) {
	que.push(start);
	visited[start] = true;
	while (!que.empty()) {
		int cur = que.front();
		que.pop();
		for (int i = 0; i < graph[cur].size(); i++) {
			int next = graph[cur][i];
			if (!visited[next]) {
				que.push(next);
				cnt += 1;
				visited[next] = true;
			}
		}
	}
	return cnt;
}

int main(void) {
	cin >> n;
	cin >> m;
	for (int i = 0; i < m; i++) {
		cin >> a >> b;
		graph[a].push_back(b);
		graph[b].push_back(a);
	}
	for (int i = 1; i <= n; i++) {
		sort(graph[i].begin(), graph[i].end());
	}
	cout << bfs(1);
}